### PR TITLE
Fix backprop mistakes + random training + update learning

### PR DESCRIPTION
**Pull request by Neavend**

### CODE REVIEW

File: neuralnet.c [hunk 1]
```diff
@@ -43,9 +48,10 @@ void train() {
   bwN3 = (double) rand() / (double) RAND_MAX;
 
   int epoch = 0;
-  while(epoch < 2000) {
+  while(epoch < 8000) {
     ++epoch;
```
Comment (alarsyo) [inline neuralnet.c]:
```diff
-while(epoch < 8000) {
-  ++epoch;
+while(epoch++ < 8000) {
```

File: neuralnet.c [hunk 2]
```diff
@@ -19,6 +19,11 @@ double output(double in[], double w[], double bw) {
   res += bw;
   return sigmoid(res);
 }
+int random_range(int min, int max)
+{
+  int diff = max-min;
+  return (int) (((double)(diff+1)/RAND_MAX) * rand() + min);
+}
```
Comment (alarsyo) [inline neuralnet.c]:
```diff
-int random_range(int min, int max)
-{
-  int diff = max-min;
-  return (int) (((double)(diff+1)/RAND_MAX) * rand() + min);
-}
+int random_range(int max)
+{
+  return rand() % max;
+}
```

File: neuralnet.c (hunk 1 above)
Comment (Neavend) [inline neuralnet.c]: 2000 * 4 = 8000
You were doing 8000 iterations in the previous code, now that we don't use two sets of while I had to increase it to 8000

Comment (Neavend) [inline neuralnet.c]: That was in the previous code and I didn't think it was necessary to refactor, you could have done that by yourself on another commit

Comment (alarsyo) [inline neuralnet.c]: You're the one who wants this merged :) I'll do it
